src/game.rs

```diff
@@ -0,0 +1,86 @@
+use crate::cars::Cars;
+
+pub struct Game {
+    pub cars: Cars,
+    pub round: u32,
+    pub total_rounds: u32,
+}
+
+pub struct RoundResult {
+    pub cars: Cars,
+}
+
+impl Game {
+    pub fn new(cars: Cars, total_rounds: u32) -> Game {
+        Game {
+            cars,
+            round: 0,
+            total_rounds,
+        }
+    }
+
+    pub fn play_all_rounds(&mut self) -> Vec<RoundResult> {
+        let mut game_results = Vec::new();
+        loop {
+            if self.is_over() {
+                break;
+            }
+            game_results.push(self.play());
+        }
+        game_results
+    }
+
+    fn play(&mut self) -> RoundResult {
+        self.cars = self.cars.play_round();
+        self.round += 1;
+        RoundResult {
+            cars: self.cars.clone(),
+        }
+    }
+
+    fn is_over(&self) -> bool {
+        self.round == self.total_rounds
+    }
+
+    pub fn get_winners(&self) -> Vec<String> {
+        let max_distance = &self.cars.calculate_max_distance();
+        let mut winners = Vec::new();
+        for car in &self.cars.value {
+            if max_distance > &car.distance {
+                continue;
+            }
+            winners.push(car.name.clone());
+        }
+        winners
+    }
+}
```

Comment:
```rs
pub fn get_winners(&self) -> Result<Vec<String>, &'static str> {
    let max_distance = &self.cars.calculate_max_distance()?;
    let mut winners = Vec::new();
    for car in &self.cars.value {
        if max_distance > &car.distance {
            continue;
        }
        winners.push(car.name.clone());
    }

    Ok(winners)
}
```

Reply:
How about handling the error inside of `get_winners` instead of returning `Result`? I tried both ways and I think that this way keeps the code at `main` cleaner.

Reply:
That's your choice.
But it looks like `eprintln!` sends error to io::stderr which make panic eventually.
I think it would be better to reduce unrecoverable error as much as we can.
Here, `calculate_max_distance` seems almost impossible to raise error so that it's find for now though.